day2/solution.go

```diff
@@ -0,0 +1,126 @@
+package day2
+
+import (
+	"github.com/piotrplaneta/adventofcode2018/utils"
+)
+
+type letterFreqencies map[rune]int
+type pair struct {
+	s1 string
+	s2 string
+}
+
+//SolvePart1 returns the answer for part 1 of day 2
+func SolvePart1() int {
+	return Checksum(adventInput())
+}
+
+//SolvePart2 returns the answer for part 1 of day 2
+func SolvePart2() string {
+	return SimilarID(adventInput())
+}
+
+//Checksum returns the product of words with 2 and 3 same letters
+func Checksum(strings []string) int {
+	listOfFrequencies := make([]letterFreqencies, len(strings))
+
+	for i, s := range strings {
+		listOfFrequencies[i] = frequencyMap(s)
+	}
+
+	return wordsWithLetterOfFrequency(listOfFrequencies, 2) * wordsWithLetterOfFrequency(listOfFrequencies, 3)
+}
+
+//SimilarID returns the same letters in similar id
+func SimilarID(strings []string) string {
+	combinations := generate2ElemCombinations(strings)
+
+	for combination := range combinations {
+		if similar(combination) {
+			return sameLetters(combination)
+		}
+	}
+
+	return ""
+}
+
+func wordsWithLetterOfFrequency(listOfFrequencies []letterFreqencies, frequency int) int {
+	result := 0
+
+	for _, v := range listOfFrequencies {
+		if containsFrequence(v, frequency) {
```

Comment:
So a built-in function for this doesn't exist in Go? I mean something similar to `Array#include?` in Ruby.

Reply:
@walerian777 not really - go does not support generics, and if you need such function for specific types (it is for Map of specific type in that case) the convention is to implement it.